conformance/utils/suite/suite.go

```diff
@@ -124,6 +124,10 @@ func (suite *ConformanceTestSuite) Setup(t *testing.T) {
 	t.Logf("Test Setup: Applying base manifests")
 	suite.Applier.MustApplyWithCleanup(t, suite.Client, suite.BaseManifests, suite.GatewayClassName, suite.Cleanup)
 
+	t.Logf("Test Setup: Applying programmatic resources")
+	secret := kubernetes.MustCreateSelfSignedCertSecret(t, "gateway-conformance-web-backend", "certificate", []string{"*"})
+	suite.Applier.MustApplyObjectsWithCleanup(t, suite.Client, []client.Object{secret}, suite.Cleanup)
```

Comment:
It might be nice to be able to attach these kinds of resource creates to the tests for which they're needed rather than having to put them all in the suite-level setup; could add `Setup`/`Teardown`-style functions to the `ConformanceTest` struct. I don't think this needs to block now, but could be a refactor to keep in mind if/when we add more of these.